File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> map;
        vector<int> ans;
        
        // Loop through input array
        for (int i = 0; i < nums.size(); ++i)
        {
            int diff = target - nums[i];
            
            if (map.find(diff) != map.end())
            {
                ans.push_back(map[diff]);
                ans.push_back(i);
                return ans;
            }
            else
            {
                map[nums[i]] = i;
            }
        }
        return ans;
    }
};

File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # Use dict to keep track of seen numbers
        # Loop through elements of nums
        # Check to see if target - current number is in dict
        # If so, return indices
        # At end of loop return false
        
        seen = {}
        for i, num in enumerate(nums):
            if target - num in seen:
                return [seen[target-num], i]
            seen[num] = i
            